### PR TITLE
Mb/pymetised

### PR DESCRIPTION
**Pull request by sesquideus**
Rewrote the internal structure of the `*Impl` classes to work with `InputSets`.

- generic and structured tests
- improved initialization script
- more recipes and associated input sets
- working with new SOFs

### CODE REVIEW

**Comment by hugobuddel** [comment]
Looks good!

I'm not sure whether we agreed upon a process. You didn't assign a reviewer, so I'm not sure what process you have in mind. Shall I/you just merge this?

File: init.sh [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #!/bin/bash
 export PYTHONPATH=.
-export SOF_DATA=~/astar/METIS_Simulations/ESO/output
+export SOF=~/astar/METIS_Pipeline_Test_Data/metis_sim_small_1/sof
+export SOF_DATA=~/astar/METIS_Pipeline_Test_Data/metis_sim_small_1/data
 export PYESOREX_PLUGIN_DIR=~/astar/METIS_Pipeline/metisp/pyrecipes/
+export PYESOREX_OUTPUT_DIR=~/astar/METIS_Pipeline_Test_Data/metis_sim_small_1/data
```
Comment (hugobuddel) [inline init.sh:6]: I think we should remove this file, or at least use the same paths as we use in the Dockerfile. But that is not really relevant to this PR though

File: metisp/pymetis/src/pymetis/recipes/img/metis_lm_img_basic_reduce.py [hunk 2]
```diff
@@ -153,7 +153,7 @@ def process_images(self) -> Dict[str, PipelineProduct]:
         return self.products
 
 
-class MetisLmBasicReduce(MetisRecipe):
+class MetisLmImgBasicReduce(MetisRecipe):
```
Comment (hugobuddel) [inline metisp/pymetis/src/pymetis/recipes/img/metis_lm_img_basic_reduce.py:156]: The `_name` should also become `metis_lm_img_basic_reduce`, but we can also change that in another PR

**Review by hugobuddel — APPROVED** [review]
Great! Thanks @sesquideus . I couldn't fully review this, but since it is clearly an improvement, let's merge this

**Comment by hugobuddel** [comment]
Note that there is technically not necessary a one-to-one correspondence between the tag, the category, and the filename. But for now this is fine.

**Comment by hugobuddel** [comment]
What is our process @sesquideus ? With @teutoburg we usually let the person who created the PR merge it.

I like that process, because usually most of the comments I give are not showstoppers, and this lets the author of the PR decide whether it is worth it to make the changes now or later.

**Comment by sesquideus** [comment]
> What is our process @sesquideus ? With @teutoburg we usually let the person who created the PR merge it.

Gladly. I think all of these are valid points but should be postponed until after the first delivery.

File: metisp/pymetis/src/pymetis/recipes/img/metis_lm_img_basic_reduce.py (hunk 2 above)
Comment (hugobuddel) [reply metisp/pymetis/src/pymetis/recipes/img/metis_lm_img_basic_reduce.py:156]: Fixed in #61

File: init.sh (hunk 1 above)
Comment (hugobuddel) [reply init.sh:6]: Fixed in #64

**Pull request merged**